app/presenters/tree_builder.rb

```diff
@@ -7,98 +7,11 @@ class TreeBuilder
-    when :images_filter    then TreeBuilderImagesFilter
-    when :vms_instances_filter    then TreeBuilderVmsInstancesFilter
-    when :templates_images_filter then TreeBuilderTemplatesImagesFilter
-
-    when :policy_simulation then TreeBuilderPolicySimulation
-    when :policy_profile    then TreeBuilderPolicyProfile
-    when :policy            then TreeBuilderPolicy
-    when :event             then TreeBuilderEvent
-    when :condition         then TreeBuilderCondition
-    when :action            then TreeBuilderAction
-    when :alert_profile     then TreeBuilderAlertProfile
-    when :alert             then TreeBuilderAlert
-
-    # reports explorer trees
-    when :db               then TreeBuilderReportDashboards
-    when :export           then TreeBuilderReportExport
-    when :reports          then TreeBuilderReportReports
-    when :roles            then TreeBuilderReportRoles
-    when :savedreports     then TreeBuilderReportSavedReports
-    when :schedules        then TreeBuilderReportSchedules
-    when :widgets          then TreeBuilderReportWidgets
-
-    # containers explorer tree
-    when :containers         then TreeBuilderContainers
-    when :containers_filter  then TreeBuilderContainersFilter
-
-    # automate explorer tree
-    when :ae               then TreeBuilderAeClass
-
-    # miq_ae_customization explorer trees
-    when :ab                    then TreeBuilderButtons
-    when :dialogs               then TreeBuilderServiceDialogs
-    when :dialog_import_export  then TreeBuilderAeCustomization
-    when :old_dialogs           then TreeBuilderProvisioningDialogs
-
-    # OPS explorer trees
-    when :diagnostics           then TreeBuilderOpsDiagnostics
-    when :rbac                  then TreeBuilderOpsRbac
-    when :servers_by_role       then TreeBuilderServersByRole
-    when :roles_by_server       then TreeBuilderRolesByServer
-    when :settings              then TreeBuilderOpsSettings
-    when :vmdb                  then TreeBuilderOpsVmdb
-
-    # PXE explorer trees
-    when :customization_templates then TreeBuilderPxeCustomizationTemplates
-    when :iso_datastores          then TreeBuilderIsoDatastores
-    when :pxe_image_types         then TreeBuilderPxeImageTypes
-    when :pxe_servers             then TreeBuilderPxeServers
-
-    # Services explorer tree
-    when :svcs                    then TreeBuilderServices
 
-    when :sa                      then TreeBuilderStorageAdapters
-
-    # Datastores explorer trees
-    when :storage     then TreeBuilderStorage
-    when :storage_pod then TreeBuilderStoragePod
-
-    when :datacenter              then TreeBuilderDatacenter
-    when :vat                     then TreeBuilderVat
-
-    when :network                 then TreeBuilderNetwork
-    when :df                      then TreeBuilderDefaultFilters
-    end
+  def self.class_for_type(type)
+    raise('Obsolete tree type.') if type == :filter
```

Comment:
darn. I did this. must have forgotten to push.
I do remember there being an invalid type flowing through.
Maybe that is all the reason more to add it